modules/nw-udn-cr.adoc

```diff
@@ -39,27 +39,74 @@ spec:
     IPAMLifecycle: Persistent # <8>
 ----
 <1> Name of your `UserDefinedNetwork` resource. This should not be `default` or duplicate any global namespaces created by the Cluster Network Operator (CNO).
-<2> The `topology` field describes the network configuration, accepted values are `Layer2` and `Layer3`. `Layer3` topology creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets. `Layer2` topology creates one logical switch shared by all nodes.
+<2> The `topology` field describes the network configuration, accepted values are `Layer2` and `Layer3`. Specifying a `Layer2` topology type creates one logical switch that is shared by all nodes.
 <3> This field specifies the topology configuration, it can be `layer2` or `layer3`.
 <4> Specifies `Primary` or `Secondary`. `Primary` is the only `role` specification supported in {product-version}.
 <5> The maximum transmission units (MTU) is an optional field. The default value is `1400`. The boundary for IPv4 is 574, and for IPv6 it is 1280.
-<6> Specifies the subnet to be used for the network across the cluster. Supports both IPv6 and dual-stack. For example, `192.168.100.0/24`,`2001:DBB::/64`. Dual-stack may set two subnets otherwise only one is allowed. When the `topology` field is set to `Layer3`, the subnet is split into smaller subnets for every node. Accepted format for subnets when `Layer3` is set are: `172.16.0.0/16/24`. For `Layer2` values in `topology` field, standard CIDR ranges are accepted. If omitted the network only provides `Layer2` communication and you must configure IP addresses.
-<7> Specifies the `subnets` used inside the OVN-Kubernetes network topology. If omitted, the OVN-Kubernetes network plugin assigns one, which is subject to change over time.
+<6> For `Layer2` topology types the following specifies config details for the `subnet` field:
++
+* The subnets field is optional.
+
+* The subnets field is of type `string` and accepts standard CIDR formats for both IPv4 and IPv6.
+
+* The subnets field accepts one or two items. For two items, they must be of a different family. For example, subnets values of `10.100.0.0/16` and `2001:db8::/64`.
+
+* Layer2 subnets may be omitted. If omitted, users must configure IP addresses for the pods. As a consequence, port security only prevents MAC spoofing.
+
+* The Layer2 `subnets` field is mandatory when `ipamLifecycle` field is specified.
+<7> It is not recommended to set this field without explicit need and understanding of the OVN network topology. Specifies the routing between different segments within the Layer 2 network. Dual-stack clusters may set 2 subnets, one for each IP family, otherwise only 1 subnet is allowed. This field is only allowed for `Primary` network. When omitted, the platform will choose a reasonable default which is subject to change over time.
```

Comment:
@tssurya not sure we want to delve deeper here ...

I feel we should at least indicate when the user would want to override this value - i.e. when they use the default join value in their infra (for whatever reason ...).

Reply:
Its ok  to me. To avoid any curiosity, I guess we can reference like for more info about default Join Subnets Please refer [this](https://docs.openshift.com/container-platform/4.16/networking/ovn_kubernetes_network_provider/configure-ovn-kubernetes-subnets.html) or just leaving as is is also fine.

Reply:
Well defaults values for UDN JoinSUbnets are https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/types/const.go#L141.

So may be we can improve callout to say avoid overlap with those.

Reply:
@anuragthehatter @ormergi thought it best to remove these fields so I did. I think they are a bit in the weeds and also might be covered better in virt use cases.